### ISSUE TITLE
[BUG] Currency conversion not occurring on Overview page in 2.38.0

### ISSUE THREAD

**Issue by nmjensen** [ghostfolio/ghostfolio#2866]
Hi there,

My base currency is AUD, and I have some stock holdings in both JPY and USD as well.

After upgrading to 2.38.0, the Overview page is treating the components in JPY and USD as if they were AUD when calculating the total portfolio value. Given there are close to 100 JPY to an AUD, it's significantly inflated my position. Which was great to look at for a second, but unfortunately is not real!

So in other words, if a holding is worth JPY ¥10,000, then it's counting it as AUD $10,000 instead of the converted value of AUD $103.

Downgrading to 2.37.0 resolves the issue.

Let me know if any further detail is needed!

Thanks,

Nick

**Comment by dtslvr** [comment]
Is there any related message about it in the log?

**Comment by nmjensen** [comment]
The log file is full of these:

> [Nest] 225  - 01/14/2024, 3:42:25 PM   ERROR [ExchangeRateDataService] No exchange rate has been found for USDAUD at 2021-08-29
[Nest] 225  - 01/14/2024, 3:42:25 PM   ERROR [ExchangeRateDataService] No exchange rate has been found for USDAUD at 2021-08-28
[Nest] 225  - 01/14/2024, 3:42:25 PM   ERROR [ExchangeRateDataService] No exchange rate has been found for USDAUD at 2021-08-27
[Nest] 225  - 01/14/2024, 3:42:25 PM   ERROR [ExchangeRateDataService] No exchange rate has been found for USDAUD at 2021-08-26
[Nest] 225  - 01/14/2024, 3:42:25 PM   ERROR [ExchangeRateDataService] No exchange rate has been found for USDAUD at 2021-08-25
[Nest] 225  - 01/14/2024, 3:42:25 PM   ERROR [ExchangeRateDataService] No exchange rate has been found for USDAUD at 2021-08-24
[Nest] 225  - 01/14/2024, 3:42:25 PM   ERROR [ExchangeRateDataService] No exchange rate has been found for USDAUD at 2021-08-23
[Nest] 225  - 01/14/2024, 3:42:25 PM   ERROR [ExchangeRateDataService] No exchange rate has been found for USDAUD at 2021-08-22
[Nest] 225  - 01/14/2024, 3:42:25 PM   ERROR [ExchangeRateDataService] No exchange rate has been found for USDAUD at 2021-08-21
[Nest] 225  - 01/14/2024, 3:42:25 PM   ERROR [ExchangeRateDataService] No exchange rate has been found for USDAUD at 2021-08-20

**Comment by dtslvr** [comment]
Can you please verify that you have all historical exchange rate data?

Go to _Admin Control_ > _Market Data_ and select the preset _Currencies_ (see _Filter by..._). There you should find `USDAUD`. If there are (red) gaps, try to gather all historical data.

**Comment by nmjensen** [comment]
There weren't any red gaps, but I've gone into USDAUD and clicked Gather Historical Data. Now those errors in the log are gone - but it's still displaying foreign currency as if it were those values in AUD.

I can't see anything in the log of note now.

Here are some screenshots when I downgrade to 2.37.0 - these are all correct.

Overview shows the correct value:
<img width="1437" alt="2 37 0" src="https://github.com/ghostfolio/ghostfolio/assets/6601052/4772c9c3-beb4-462c-a3dc-6e33eef1b2bb">

Holdings show the correctly converted value of the shares held in foreign currencies, converted to AUD:
<img width="1434" alt="2 37 0-holdings" src="https://github.com/ghostfolio/ghostfolio/assets/6601052/a50eb146-23a8-48ca-a439-f49e092fbc74">

But then, when I upgrade to latest, 2.38.0, Overview shows an incorrect value:
<img width="1428" alt="2 38 0" src="https://github.com/ghostfolio/ghostfolio/assets/6601052/c65e42ea-abe0-42b1-99f4-bff03ed59105">

Note in the below East Japan Railway Company, iShares Core Nikkei 225 ETF (both in JPY) and Netflix, Inc (in USD) are showing the wrong values - these are the foreign currency values but being displayed as AUD:
<img width="1433" alt="2 38 0-holdings" src="https://github.com/ghostfolio/ghostfolio/assets/6601052/9e5094f3-8648-4567-a290-af008dcd188e">

When I click on one of them, it oddly then shows the correctly converted value:
<img width="1433" alt="2 38 0-9020" src="https://github.com/ghostfolio/ghostfolio/assets/6601052/b1c5ea18-7d0c-4351-a58c-1410fdbf95f7">

When I downgrade back to 2.37.0 everythings displays as normal again (the first set of screenshots).


Cheers,

Nick

**Comment by dtslvr** [comment]
Thank you for the overview.

I have tried to reproduce the scenario, but unfortunately it looks fine.

1. Create new user account
2. Set _Base Currency_ to `AUD`
3. Import activity ([ghostfolio-export-9020.T.json](https://github.com/ghostfolio/ghostfolio/files/13930619/ghostfolio-export-9020.T.json))
4. Overview shows `87.65 AUD`

Can you describe how to reproduce?

**Comment by nmjensen** [comment]
I've tried it with another holding, and these are the steps that I can use to recreate. Does the same happen to you? Or is it something specific to me?

1. Downgrade to 2.37.0
Overview displays a value of AUD $24,690.76:
<img width="1431" alt="2-2 37 0-overview" src="https://github.com/ghostfolio/ghostfolio/assets/6601052/61847dc4-017f-48a3-9826-ac3109e6c739">

2. Add a new Tokyo Stock Exchange trade through Overview > Holdings > Manage Activities > Add, with the following details:
- Type: Buy
- Account: Selected my JPY share trading account
- Name, symbol or ISN: 7974.t
- Date: 14 Jan 2024
- Quantity: 100
- Price: 8125

3. Go back to the overview. This is displaying correctly as AUD $33,066.38 (100 shares at JPY ¥8125 = JPY ¥812,500 = AUD $8,375.62. Original portfolio value prior to trade was AUD $24,690.76, so AUD $24,690.76 + AUD $8,375.62 = AUD $33,066.38.
<img width="1434" alt="2-2 37 0-overview_2nd" src="https://github.com/ghostfolio/ghostfolio/assets/6601052/26cc1c88-9362-4620-b0e9-ec8ab79f2abb">

4. Now, I upgrade to 2.38.0 without any other changes. The Overview is now displaying a portfolio value of AUD $1,786,466.81 (which is AUD $812,500 more than it displays if I upgrade without this extra JPY ¥812,500 trade.
<img width="1433" alt="2-2 38 0-overview" src="https://github.com/ghostfolio/ghostfolio/assets/6601052/9938862c-4e57-4a33-b213-8f99f0a75cc2">

5. I downgrade to 2.37.0 without any other changes. The Overview now displays the correct AUD value again:
<img width="1434" alt="2-2 37 0-overview_after_downgrade" src="https://github.com/ghostfolio/ghostfolio/assets/6601052/bdfc364c-42ba-4d25-9b09-e24fd918f809">


Cheers,

Nick

**Comment by dtslvr** [comment]
I cannot recreate this scenario locally. Can you reproduce it in the [cloud](https://ghostfol.io) and share the account with me? Thank you.

**Comment by cbirkenbeul** [comment]
I've got the exact same error with EUR instead of AUD. Same error message in Logs. 

I've also tried to flush the cache db but got an error message in the log file
`│ main [Nest] 198  - 01/14/2024, 10:14:51 AM   ERROR [ExceptionsHandler] ERR unknown command 'flushdb', with args beginning with:                    │
│ main ReplyError: ERR unknown command 'flushdb', with args beginning with:                                                                          │
│ main     at parseError (/ghostfolio/apps/api/node_modules/redis-parser/lib/parser.js:179:12)                                                       │
│ main     at parseType (/ghostfolio/apps/api/node_modules/redis-parser/lib/parser.js:302:14)`

**Comment by dtslvr** [comment]
I have released a new version with a potential fix ([Ghostfolio 2.40.0](https://github.com/ghostfolio/ghostfolio/releases/tag/2.40.0)). Can you please try out and provide feedback?

**Comment by cbirkenbeul** [comment]
For me it’s fixed. Thanks for quick bugfix. 👍🏻

**Comment by fabro21** [comment]
Hi there seems to be a bug not letting the historical data to go beyond 3800 days

<img width="1159" height="383" alt="Image" src="https://github.com/user-attachments/assets/c21e479f-fb44-4329-b0e9-77c221d8d5f3" />

When you have activities beyond such range, this triggers the currency exchange rate error

<img width="1442" height="451" alt="Image" src="https://github.com/user-attachments/assets/007e2efe-05fd-430b-b89a-c0524f42ddb2" />
